app/controllers/expenses_controller.rb

```diff
@@ -1,9 +1,11 @@
 class ExpensesController < ApplicationController
-
   def create
-    week = current_user.current_week
-    week.expenses.create(amount: params[:expenses][:amount], manner: params[:expenses][:type])
-    render json: { week: week }
+    current_week = current_user.find_week(0.week)
+    next_week = current_user.find_week(1.week) if params[:expenses][:week] == "true"
+    if params[:expenses][:amount].present?
+      week = next_week || current_week
+      week.expenses.create(amount: params[:expenses][:amount], manner: params[:expenses][:type])
+    end
+    render json: { week: current_week }
   end
```

Comment:
I think this method would read better if we started by finding the exact week we want to manipulate - The amount check is nice but would make more sense as a model level (the `validates amount, presence: true` you added)

```ruby

def create
  week = params[:expenses][:next_week] == 'true' ? get_next_week : get_this_week
  week.expenses.create(amount: params[:expenses][:amount], manner: params[:expenses][:type])
end
```
`get_next_week` and `get_this_week` can be protected methods in this controller, that get the appropriate week object.